equinix/provider.go

```diff
@@ -171,31 +168,20 @@ func Provider() *schema.Provider {
 
 func configureProvider(ctx context.Context, d *schema.ResourceData, p *schema.Provider) (interface{}, diag.Diagnostics) {
 	mrws := d.Get("max_retry_wait_seconds").(int)
+    rt := d.Get("request_timeout").(int)
 
 	config := Config{
-		AuthToken:    d.Get("auth_token").(string),
-		MaxRetries:   d.Get("max_retries").(int),
-		MaxRetryWait: time.Duration(mrws) * time.Second,
-	}
-	if v, ok := d.GetOk("endpoint"); ok {
-		config.BaseURL = v.(string)
-	}
-	if v, ok := d.GetOk("client_id"); ok {
-		config.ClientID = v.(string)
-	}
-	if v, ok := d.GetOk("client_secret"); ok {
-		config.ClientSecret = v.(string)
+		AuthToken:      d.Get("auth_token").(string),
+		BaseURL:        d.Get("endpoint").(string),
+		ClientID:       d.Get("client_id").(string),
+		ClientSecret:   d.Get("client_secret").(string),
+		Token:          d.Get("token").(string),
+		RequestTimeout: time.Duration(rt) * time.Second,
+		PageSize:       d.Get("response_max_page_size").(int),
```

Comment:
It looks like `PageSize` this can still be `nil`. Should we define a default?

Reply:
Hmm. That said, the acceptance tests didn't have any problem with this.

Reply:
That Get function calls [ValueOrZero](https://github.com/hashicorp/terraform-plugin-sdk/blob/v1.17.2/helper/schema/field_reader.go#L37) internally to ensure a consistent non-nil return value. In case of `int` it returns `0`